source/admin_manual/mailbox_formats/Maildir.rst

```diff
@@ -0,0 +1,415 @@
+|                                                                                                   +----------------------------------------------------------+
+|                                                                                                   | **T** = trashed                                          |
+|                                                                                                   +----------------------------------------------------------+
+|                                                                                                   | **l** = IMAP tag #12 (0=a, 1=b, 2=c, etc) as defined in  |
+|                                                                                                   | that folder's ``dovecot-keywords`` file.                 |
+|                                                                                                   +----------------------------------------------------------+
+|                                                                                                   | **n** = IMAP tag #14 (0=a, 1=b, 2=c, etc) as defined in  |
+|                                                                                                   | that folder's ``dovecot-keywords`` file.                 |
++---------------------------------------------------------------------------------------------------+----------------------------------------------------------+
+
+.. _`IMAP INTERNALDATE, RFC 3501 [2.3.3]`: https://tools.ietf.org/html/rfc3501#section-2.3.3
+.. _`index files`: https://wiki.dovecot.org/IndexFiles
+
+.. _`Maildir and Filesystems`:
+
+Maildir and Filesystems
+^^^^^^^^^^^^^^^^^^^^^^^
+
+General Comparisons of Maildir on Different Filesystems
+-------------------------------------------------------
+
+* https://www.thesmbexchange.com/eng/qmail_fs_benchmark.html
+* https://www.htiweb.inf.br/benchmark/fsbench.htm (including some graphs)
+
+Linux ext2 / ext3 
+-----------------
+
+The main disadvantage in using these filesystems is that searching can be
+slightly slower, and access to very large mailboxes (thousands of messages)
+can get slow with filesystems which don't have directory indexes.
+
+Old versions of ext2 and ext3 on Linux don't support directory indexing (to
+speed up access), but newer versions of ext3 do, although you may have to
+manually enable it. You can check if the indexing is already enabled with
+``tune2fs``:
+
+.. code-block:: none
+
+  tune2fs -l /dev/hda3 | grep features
+
+If you see ``dir_index``, you're all set. If ``dir_index`` is missing, add it
+using:
+
+.. code-block:: none
+
+  umount /dev/hda3
+  tune2fs -O dir_index /dev/hda3
+  e2fsck -fD /dev/hda3
+  mount /dev/hda3
+
+ReiserFS
+--------
+
+ReiserFS was built to be fast with lots of small files, so it works well with
+Maildir.
```

Comment:
ReiserFS is pretty dead these days?

Reply:
Heh. Yeah, I don't think reiserfs is a reasonable choice these days compared to other options.  Removed.